h2o-r/h2o-package/R/kvstore.R

```diff
@@ -210,6 +210,10 @@ h2o.getModel <- function(model_id) {
     }
   })
 
+  # Store base models for the Stacked Ensemble in user-readable form
+  if (json$algo == "stackedensemble"){
+    model$base_models <- unlist(lapply(parameters$base_models, function (base_model) base_model$name))
+  }
```

Comment:
yeah, that's exactly what I was afraid of with this approach of adding properties to `model` slot.
It's opening the door to "put any specific logic here" and clutter the algo's logic this file and their dedicated one.

Do you feel like adding a hook here?
At this point, you know the `algo`, you have the `model`, the user params (`parameters`) and all params (`allParams`): what about checking if there is a private method `.h2o.fill_{algo}` and call it if present, e.g.
```
fill_func = paste0(".h2o.fill_", algo)
if (exists(fill_func, mode="function") {
   do.call(fill_func, list(model, parameters, allparams))
}
```
This way you can define 
```
.h2o.fill_stackedensemble <- function(model, user_params, all_params) { ... }
```
in the `stackedensemble.R` file without polluting the `kvstore.R`.

Didn't really try it, but would be nice if it worked

Reply:
I will add the hook there. There will be just a slight change:
```
fill_func = paste0(".h2o.fill_", algo)
if (exists(fill_func, mode="function") {
  model <- do.call(fill_func, list(model, parameters, allparams))
}
```

because AFAIK R likes its function arguments immutable. I could do "magic" to get the parent environment  to modify the values I want from the caller environment but this seems nicer. If you have better idea I will be happy to hear it because this solution has a problem of being able to modify only the model (which might not be all that we will want to do). I could also return all three arguments and update all of them but I don't know if that's what we want...

Reply:
ok, great. I think just returning `model` is fine for now, and should cover the majority of customizations we might want to do.